src/grt/src/MakeWireParasitics.cpp

```diff
@@ -114,54 +128,154 @@ sta::Pin* MakeWireParasitics::staPin(Pin& pin)
+        parasitics_->incrCap(n1, 0, analysis_point_);
+        debugPrint(logger_,
+                   GRT,
+                   "est_rc",
+                   1,
+                   "(node-to-wire) {} -> {} via r={}",
+                   parasitics_->name(node),
+                   parasitics_->name(n1),
+                   units->resistanceUnit()->asString(cut_layer->getResistance()));
+      }
+      on_grid_node_map_.erase(n1_pt);
+
+      for (auto& node : on_grid_node_map_[n2_pt]) {
+        parasitics_->incrCap(node, 0, analysis_point_);
+        parasitics_->makeResistor(nullptr, node, n2, cut_layer->getResistance(), analysis_point_);
+        parasitics_->incrCap(n2, 0, analysis_point_);
+        debugPrint(logger_,
+                   GRT,
+                   "est_rc",
+                   1,
+                   "(node-to-wire) {} -> {} via r={}",
+                   parasitics_->name(node),
+                   parasitics_->name(n2),
+                   units->resistanceUnit()->asString(cut_layer->getResistance()));
+      }
+      on_grid_node_map_.erase(n2_pt);
+
+      routes_cnt++;
+    }
+  }
+}
+
+void MakeWireParasitics::makeViaParasitics(odb::dbNet* net,
+                       std::vector<GSegment>& routes,
+                       int& routes_cnt)
+{
+  sta::Units* units = sta_->units();
+  for (GSegment& route : routes) {
+    int wire_length_dbu = route.length();
+    if (wire_length_dbu == 0) {
+      int lower_layer = min(route.init_layer, route.final_layer);
+      odb::dbTechLayer* cut_layer
+          = tech_->findRoutingLayer(lower_layer)->getUpperLayer();
+      float res = cut_layer->getResistance();  // assumes single cut
+      float cap = 0.0;
+
+      sta::ParasiticNode* n1
+          = ensureParasiticNode(route.init_x, route.init_y, route.init_layer);
+      sta::ParasiticNode* n2
+          = ensureParasiticNode(route.final_x, route.final_y, route.final_layer);
+            
+      RoutePt n1_pt(route.init_x, route.init_y, route.init_layer);
+      RoutePt n2_pt(route.final_x, route.final_y, route.final_layer);
+      std::vector<sta::ParasiticNode*>& nodes1 = on_grid_node_map_[n1_pt];
+      std::vector<sta::ParasiticNode*>& nodes2 = on_grid_node_map_[n2_pt];
+      if (!nodes1.empty()) {
+        for (sta::ParasiticNode* node : nodes1) {
+          parasitics_->incrCap(node, cap / 2.0, analysis_point_);
+          parasitics_->makeResistor(nullptr, node, n2, res, analysis_point_);
+          parasitics_->incrCap(n2, cap / 2.0, analysis_point_);
+          debugPrint(logger_,
+                     GRT,
+                     "est_rc",
+                     1,
+                     "(node-to-via) {} -> {} via r={}",
+                     parasitics_->name(node),
+                     parasitics_->name(n2),
+                     units->resistanceUnit()->asString(res));
+        }
+      } 
+      // connect the pin nodes located in n2 to n1
+      if (!nodes2.empty()) {
```

Comment:
There is no point in testing the collection for empty because the loop will not do anything if it is empty